libs/langchain/langchain/utils/utils.py

```diff
@@ -2,13 +2,38 @@
 import contextlib
 import datetime
 import importlib
+import os
 from importlib.metadata import version
-from typing import Any, Callable, Optional, Set, Tuple
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple
 
 from packaging.version import parse
 from requests import HTTPError, Response
 
 
+def get_from_dict_or_env(
```

Comment:
ooc whyre these being added?

Reply:
- [X]

Sorry. I should be more cautious.